hazelcast/src/main/java/com/hazelcast/jet/JetService.java

```diff
@@ -299,4 +300,18 @@ default <T> Observable<T> newObservable() {
      */
     @Nonnull
     Collection<Observable<?>> getObservables();
+
+    /**
+     * Associates the given listener with the given jobIds.
+     *
+     * @since 5.3
```

Comment:
We should either implement some kind of listener deregistration when a job completes, or transfer the burden to the user by specifying the need to remove the listener after the job completes.